src/hmem.c

```diff
@@ -50,6 +50,8 @@ struct ofi_hmem_ops {
 	int (*get_handle)(void *dev_buf, void **handle);
 	int (*open_handle)(void **handle, uint64_t device, void **ipc_ptr);
 	int (*close_handle)(void *ipc_ptr);
+	int (*host_register)(void *ptr, size_t size);
+	int (*host_unregister)(void *ptr);
```

Comment:
PR #6218 wants to add a device register call.  And now I think we're overloading what 'register' means to the point that I can't keep track.  Maybe we should take this to the mailing list to make sure that we align on a useful abstraction and not device/interface abstractions that are not really abstractions but only needed to support a single GPU vendor.

Reply:
update: PR #6218 was updated to remove the device register call.  So I think we proceed here with these changes as they are and revisit later if needed.

Reply:
I think the host register makes sense since much GPU vendors have a similar concept. The device register abstraction, which I admit was my suggestion, may only be used for the CUDA and GDR copy use case.

Reply:
Sound good.